packages/react/toolbar/src/Toolbar.tsx

```diff
@@ -0,0 +1,246 @@
+import * as React from 'react';
+import { RovingFocusGroup, useRovingFocus } from '@radix-ui/react-roving-focus';
+import { Primitive } from '@radix-ui/react-primitive';
+import { getSelector } from '@radix-ui/utils';
+import { createContext, composeEventHandlers } from '@radix-ui/react-utils';
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+import type { Merge } from '@radix-ui/utils';
+
+/* -------------------------------------------------------------------------------------------------
+ * Toolbar
+ * -----------------------------------------------------------------------------------------------*/
+
+const TOOLBAR_NAME = 'Toolbar';
+const TOOLBAR_CONTEXT_NAME = TOOLBAR_NAME + 'Context';
+const TOOLBAR_DEFAULT_TAG = 'div';
+
+type Orientation = React.AriaAttributes['aria-orientation'];
+type Direction = 'ltr' | 'rtl';
+
+export type RovingFocusGroupOptions = {
+  orientation?: Orientation;
+  dir?: Direction;
+  loop?: boolean;
+};
+
+type ToolbarOwnProps = Merge<
+  Polymorphic.OwnProps<typeof Primitive>,
+  {
+    /**
+     * The orientation of the toolbar.
+     * Mainly so arrow navigation is done accordingly (left & right vs. up & down)
+     * (default: horizontal)
+     */
+    orientation?: React.AriaAttributes['aria-orientation'];
+    /**
+     * The direction of navigation between toolbar items.
+     * (default: ltr)
+     */
+    dir?: Direction;
+    /**
+     * Whether keyboard navigation should loop focus
+     * (default: true)
+     */
+    loop?: boolean;
+  }
+>;
+
+type ToolbarPrimitive = Polymorphic.ForwardRefComponent<
+  typeof TOOLBAR_DEFAULT_TAG,
+  ToolbarOwnProps
+>;
+
+type ToolbarContextValue = {
+  name: typeof TOOLBAR_CONTEXT_NAME;
+  orientation: Orientation;
+};
+
+const [ToolbarContext, useToolbarContext] = createContext<ToolbarContextValue>(
+  TOOLBAR_CONTEXT_NAME,
+  TOOLBAR_NAME
+);
+
+const Toolbar = React.forwardRef((props, forwardedRef) => {
+  const {
+    children,
+
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
+    'aria-controls': ariaControls,
```

Comment:
Maybe I am missing something, admittedly it is late here, but I can' find where is that mentioned? I read through: 
* https://www.w3.org/TR/wai-aria-practices-1.1/#toolbar
* https://www.w3.org/TR/wai-aria-1.1/#toolbar

Moreover, this will make it very annoying to use the Toolbar in a situation where no specific HTML element is the target of the Toolbar tools.

Let me know if I am wrong.

Reply:
I saw that they used it here https://w3c.github.io/aria-practices/examples/toolbar/toolbar.html

Accessibility-wise I would think a toolbar should be controlling something. If it's not, then I would have thought consumer should probably reach for a `menubar` instead (which is also on our todo list) https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html

@benoitgrelard Thoughts here?

Reply:
* Yes the example use it, but nowhere it says it is required, so we know consumers would notice the difference from their previous WAI ARIA knowledge.
* `menubar` can't be a substitute because the expectations are very specific and doesn't work will with the use case of grouping some tools and buttons together: 
  - Keyboard interactions: https://www.w3.org/TR/wai-aria-practices/#keyboard-interaction-12
  - Function: "...provides site navigation menus" https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html
* Consider the following scenario: adding a toolbar that contains `select` components to choose among other things the `language` of the website, should that target the root container of the website? 
* I would say that is unnessecary but I wouldn't die on this hill either, so waiting for @benoitgrelard  thoughts too.

Reply:
> so we know consumers would notice the difference from their previous WAI ARIA knowledge.

Our primitives specifically aim to provide as much accessibility as possible out of the box so that our consumers do not need to think about it. Some of our consumers are more junior and do not have WAI ARIA knowledge.

I know the example says it isn't required but I thought a `toolbar` doesn't really make sense if it doesn't control _something_ so we would enforce the correct accessibility here.

> `menubar` can't be a substitute because the expectations are very specific and doesn't work will with the use case of grouping some tools and buttons together

It sounds like a `menubar` does work for that:

> The toolbar is often a subset of functions found in a `menubar`
> https://www.w3.org/TR/wai-aria-1.1/#toolbar 

> A menu bar is used to create a consistent set of frequently used commands 
> https://www.w3.org/TR/wai-aria-1.1/#menubar

Also, here is an example of a `menubar` controlling an editor:

https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-2/menubar-2.html

> Consider the following scenario: adding a toolbar that contains `select` components to choose among other things the language of the website, should that target the root container of the website?

I believe that example would be a `menubar`.

I'm also keen to hear @benoitgrelard's thoughts because I am no expert here and I'm also happy to go without it if that is consensus. It does feel strange to have a set of "tools" that control nothing though 🙂

Reply:
I stand corrected! now I share your opinion and to your point adding `aria-controls` when it is not needed is a better outcome than forgetting to add one when it is needed.

Reply:
Indeed, I'd personally prefer that the consumer explicitly passed `arial-controls={null}` if they knew what they were doing and **didn't** want it to control anything. It might be worth updating the types to accept a `null` value.

Reply:
@Nazeh, Soooo.... I really wasn't sure on this one so reached out to the a11y community on Slack and they have said that the `toolbar` itself wouldn't have the `aria-controls`. Instead, the controls within it would optionally have it.

So... thanks for the discussion but I think we can leave it optional like we had it 🙈  Sorry!

I also have learnt that I did indeed misinterpret the `toolbar` docs here:

> The toolbar is often a subset of functions found in a `menubar`
> w3.org/TR/wai-aria-1.1/#toolbar

Apparently that doesn't mean a toolbar goes inside a menubar.  Rather, it means that some functions will be found inside a toolbar, and inside a menubar. 

Apologies! Feel free to revert any changes.

Reply:
I think that all makes sense to me, sorry I'm late to the party! 😀 
Probably no need to even do anything about `aria-controls` on `ToolbarItem` because it's already a DOM prop anyway.

Reply:
@benoitgrelard Sure, but for things like this, my thinking was that it might be worth adding for documentation purposes so we can guide people on accessibility practises in their editors. 

I think we can probably revisit that later for all of our components though if we think it's worthwhile so happy to leave it out here :+1: